### ISSUE TITLE
Build fails: error: non-const lvalue reference to type 'char **' cannot bind to a value of unrelated type 'const char **'

### ISSUE THREAD

**Issue by yurivict** [QuantStack/xeus-cling#178]
```
[1/2] /usr/bin/c++  -DGUID_LIBUUID -DLLVM_DIR=\"/usr/local/llvm60\" -DXCPP_TAGCONFS_DIR=\"/usr/local/etc/xcpp/tags.d\" -DXCPP_TAGFILES_DIR=\"/usr/local/share/xcpp/tagfiles\" -Dxeus_cling_EXPORTS -I/usr/ports/devel/xeus-cling/work/xeus-cling-0.4.5/include -isystem /usr/local/include -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -fno-omit-frame-pointer -Wunused-parameter -Wextra -Wreorder -march=native -std=c++14 -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -fno-omit-frame-pointer   -Wunused-parameter -Wextra -Wreorder -march=native -std=gnu++11 -MD -MT CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o -MF CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o.d -o CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o -c /usr/ports/devel/xeus-cling/work/xeus-cling-0.4.5/src/xoptions.cpp
FAILED: CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o 
/usr/bin/c++  -DGUID_LIBUUID -DLLVM_DIR=\"/usr/local/llvm60\" -DXCPP_TAGCONFS_DIR=\"/usr/local/etc/xcpp/tags.d\" -DXCPP_TAGFILES_DIR=\"/usr/local/share/xcpp/tagfiles\" -Dxeus_cling_EXPORTS -I/usr/ports/devel/xeus-cling/work/xeus-cling-0.4.5/include -isystem /usr/local/include -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -fno-omit-frame-pointer -Wunused-parameter -Wextra -Wreorder -march=native -std=c++14 -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -fno-omit-frame-pointer   -Wunused-parameter -Wextra -Wreorder -march=native -std=gnu++11 -MD -MT CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o -MF CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o.d -o CMakeFiles/xeus-cling.dir/src/xoptions.cpp.o -c /usr/ports/devel/xeus-cling/work/xeus-cling-0.4.5/src/xoptions.cpp
/usr/ports/devel/xeus-cling/work/xeus-cling-0.4.5/src/xoptions.cpp:32:36: error: non-const lvalue reference to type 'char **' cannot bind to a value of unrelated type 'const char **'
        return parent::parse(argc, argv);
                                   ^~~~
/usr/local/include/cxxopts.hpp:1642:35: note: passing argument to parameter 'argv' here
Options::parse(int& argc, char**& argv)
                                  ^
1 error generated.
```

cling version 0.5-2018-08-13
xeus-cling version 0.4.5
FreeBSD 11.2

**Comment by SylvainCorlay** [comment]
Surprised that we did not hit this, or we may be using slightly different version of cxxopts.

Does the following change in `xoptions.cpp` fix the issue for you?

```diff
        int argc = copt_strings.size();
-        auto argv = &copt_strings[0];
+        const auto argv = &copt_strings[0];
        return parent::parse(argc, argv);
```

**Comment by SylvainCorlay** [comment]
Ok, I can reproduce:

This comes from a backward-incompatible change in cxxopts, between `2.1.0` and `2.1.1`. We had not packaged `2.1.1` yet, and therefore not encountered the bug yet.

I am on it.

**Comment by SylvainCorlay** [comment]
Fixed! I released `xeus-cling` 0.4.6 with the workaround.
